docs/reference/components/amqp.cue

```diff
@@ -0,0 +1,116 @@
+package metadata
+
+components: _amqp: {
+	features: {
+		collect: from: {
+			service: services.amqp
+			interface: {
+				socket: {
+					api: {
+						title: "Amqp protocol"
+						url:   urls.amqp_protocol
+					}
+					direction: "incoming"
+					port:      5672
+					protocols: ["tcp"]
+					ssl: "optional"
+				}
+			}
+		}
+
+		send: to: {
+			service: services.amqp
+			interface: {
+				socket: {
+					api: {
+						title: "Amqp protocol"
+						url:   urls.amqp_protocol
+					}
+					direction: "outgoing"
+					protocols: ["tcp"]
+					ssl: "optional"
+				}
+			}
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	configuration: {
```

Comment:
We do have some precedent for just letting the user specify the full connection URL in the config. For example, see https://vector.dev/docs/reference/configuration/sources/postgresql_metrics/ . I could see allowing the same here rather than separating out the options.

Reply:
I go back and forth on this one, especially since connection timeout is a query parameter in the url. 

If it's ok, I think I want to leave this one split out.

Reply:
True. I guess, unless you are opposed, I'd lean towards the connection string for consistency with some of our other components and the fact that you are building a connection string out of these parts anyway.

Reply:
Switched to a single connection string